client/network/src/service.rs

```diff
@@ -104,9 +104,9 @@ pub struct NetworkService<B: BlockT + 'static, H: ExHashT> {
 	/// Number of peers we're connected to.
 	num_connected: Arc<AtomicUsize>,
 	/// The local external addresses.
-	external_addresses: Arc<Mutex<Vec<Multiaddr>>>,
+	external_addresses: Arc<Mutex<HashSet<Multiaddr>>>,
```

Comment:
Why not use the newly introduced `ExternalAddresses` here too?

Reply:
I find it more intelligible that this piece of code cannot call `on_swarm_event` on `ExternalAddresses`